### PR TITLE
lots of small tweaks and cleanup

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: k8s/__init__.py [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
-            (
-                i
-                for i in response["items"]
-                if i["metadata"]["annotations"].get("juju.io/unit") == juju_unit
-            ),
-            None,
-        )
+        for i in response["items"]:
+            if i["metadata"]["annotations"].get("juju.io/unit") == juju_unit:
+                status_dict = i
```
Comment (facundobatista) [inline k8s/__init__.py:28]: extra points for using `item` or something, not `i`

File: k8s/__init__.py [hunk 2]
```diff
@@ -36,11 +38,13 @@ class APIServer:
     the pod.
     """
 
+    _dir = Path("/var/run/secrets/kubernetes.io/serviceaccount")
```
Comment (facundobatista) [inline k8s/__init__.py:41]: Yes, it's a dir. What dir? Maybe using a name like `_account_dir` or  `_service_account_dir` would improve readability.

File: setup.py [hunk 3]
```diff
@@ -1,26 +1,55 @@
...
 with open("requirements.txt", "rt", encoding='utf8') as fh:
     requirements = fh.read().split('\n')
 
+version = _get_version()
+version_path = Path("k8s/version.py")
+version_backup = Path("k8s/version.py~")
+version_path.rename(version_backup)
+try:
+    with version_path.open("wt", encoding="utf8") as fh:
+        fh.write('''\
```
Comment (facundobatista) [inline setup.py:30]: Please put this text as a "template" or something at module level, so it doesn't break visual indentation here, thanks!

File: k8s/__init__.py (hunk 2 above)
Comment (chipaca) [reply k8s/__init__.py]: what do you think of the new approach? does it end up being more readable as used?

Comment (facundobatista) [reply k8s/__init__.py]: I prefer doing `_dir / "hola.txt` (or `_path / "hola.txt"`), but my point is that it shouldn't be called `_dir` or `_path` (as this definition is out of context and will be used lot of lines below).  What about calling it `_account_dir`, for example?

File: setup.py [hunk 4]
```diff
@@ -1,26 +1,57 @@
...
+    with version_path.open("wt", encoding="utf8") as fh:
+        fh.write(
+            dedent(
+                """
+            # this is a generated file
+
+            version = {!r}
+            """
+            ).format(version)
+        )
```
Comment (facundobatista) [inline setup.py:39]: The indentation in line 33 and dedentation in 34 (leaving the text at the same column that the `dedent` call is weird. 

What about?

```
        fh.write(
            dedent("""
                # this is a generated file
                version = {!r}
            """).format(version)
        )
```

(but it's ok if you like it more this way)

**Pull request merged**